### DISCUSSION TITLE
Grouping statistics by periods

### DISCUSSION THREAD

**Discussion by devopg** [MaterializeInc/materialize#12716] [Q&A] [answered]
Hello. I’m a newbie in Materialize . I am inspired by this software and I’m going to learn all the documentation 😅

I need to make statistics on periods (day|week|month|year) for example like this (but not the percentages , but the values):
![image](https://user-images.githubusercontent.com/61819835/170440990-0f31843d-5370-4a18-9733-bc4d7222c74f.png)

What is the optimal query I should make ?
I didn't find the LAG, LEAD function for materialize views in the documentation to take the last value of last month.

**Comment by bobbyiliev** [comment, marked as answer]
Hi there @devopg,

We did have a tracking issue for adding support for`LEAD()` and `LAG()` here: https://github.com/MaterializeInc/materialize/issues/10601

This is already fixed in https://github.com/MaterializeInc/materialize/pull/11891, https://github.com/MaterializeInc/materialize/pull/11785 but note that those will be available in future releases.

In the meantime as suggested [here](https://materializecommunity.slack.com/archives/C015KDVS7EV/p1641508811044400), you can use a [temporal filter](https://materialize.com/docs/guides/temporal-filters/) to create a view that has a sliding window of the last 24 hours of data and etc. this temporal filter will evolve continuously, and keep a finite amount of state, while elegantly sliding with time.

We've also had a similar discussion regarding window functions like `LAG()` and `LEAD()` in our community Slack [here](https://materializecommunity.slack.com/archives/C014S33T01M/p1642719051044000):

> Window functions are difficult in incremental systems because one update can produce a huge amount of changes in output. You can get the same functionality as many of the window functions using an idiom called "Top K by Group" documented here https://materialize.com/docs/guides/top-k/

Here is also another interesting read that you might find helpful for your usecase:

https://materialize.com/slicing-up-temporal-aggregates-in-materialize/

Hope that this helps!

**Reply by devopg** [reply]
okey cool , will be w8 future releases

**Reply by bobbyiliev** [reply]
Hi there @devopg,

The `lead`/`lag` functions are now available in v0.26.3. For the complete release notes you can visit the documentation here:

https://materialize.com/docs/release-notes/#v0.26.3
